R/init.R

```diff
@@ -122,52 +119,56 @@ init <- function(data,
       )
     )
   }
-
+  if (is.list(data) && !inherits(data, "teal_data_module")) {
+    checkmate::assert_list(data, names = "named")
+    data <- do.call(teal.data::teal_data, data)
+  }
   checkmate::assert_multi_class(data, c("teal_data", "teal_data_module"))
-  checkmate::assert_multi_class(modules, c("teal_module", "list", "teal_modules"))
+
+  ## `modules`
+  if (inherits(modules, "teal_module")) {
+    modules <- list(modules)
+  }
+  if (inherits(modules, "list")) {
+    modules <- do.call(teal::modules, modules)
+  }
+  checkmate::assert_class(modules, "teal_modules")
+
+  ## `filter`
   checkmate::assert(
     checkmate::check_class(filter, "teal_slices"),
```

Comment:
Nitpicking, but consider manually adding  `.var.name` when performing a generic `assert` on the same variable. (individual messages have it though)

![image](https://github.com/insightsengineering/teal/assets/211358/fefd4fd4-dbf9-480d-9f36-ad5b45677c8f)

```suggestion
    .var.name = "filter",
    checkmate::check_class(filter, "teal_slices"),
```